app/services/generic_events/runner.rb

```diff
@@ -4,9 +4,9 @@ def initialize(eventable)
       @eventable = eventable
     end
 
-    def call
-      @eventable.generic_events.applied_order.each(&:trigger)
-      @eventable.handle_event_run
+    def call(dry_run: false)
+      @eventable.generic_events.applied_order.each { |e| e.trigger({ dry_run: dry_run }) }
```

Comment:
Not sure about passing arguments as a hash here - it might not even be correctly determined as a named parameter and could be running the events without setting dry run mode. Really needs a spec to be sure I think 🤔 

```ruby
@eventable.generic_events.applied_order.each { |e| e.trigger(dry_run: dry_run) }
```

Reply:
sorry - this is wip which accidentally got pushed - ignore fo now

Reply:
hopefully now tidied up

Reply:
Yep, looks better now 👍